### PR TITLE
Swift4 migration

### PR DESCRIPTION
**Pull request by JettF**
Migrated to swift 4.

### CODE REVIEW

File: Source/ResponseSerialization.swift [hunk 1]
```diff
@@ -61,7 +61,7 @@ extension DataRequest {
                     let failureReason = "XML could not be serialized into response object: \(xml)"
                     let userInfo: [AnyHashable: Any] = [NSLocalizedFailureReasonErrorKey: failureReason]
                     let errorCode = AFError.responseSerializationFailed(reason: .inputDataNil)._code
-                    let error = NSError(domain: S3Error.Domain, code: errorCode, userInfo: userInfo)
+                    let error = NSError(domain: S3Error.Domain, code: errorCode, userInfo: userInfo as! [String : Any])
```
Comment (AnthonyMDev) [inline Source/ResponseSerialization.swift]: Generally, using explicit casting (!) is a bad idea. Can't we just make ` userInfo` type `[String: Any]` on line 62?

**Review by AnthonyMDev — CHANGES_REQUESTED** [review]

File: Source/ResponseSerialization.swift (hunk 1 above)
Comment (JettF) [reply Source/ResponseSerialization.swift]: Are you looking at an older commit here? Cause when I opened the file to make the change it was already done.

Comment (JettF) [reply Source/ResponseSerialization.swift]: This was fixed in commit 397a93c.

**Comment by av0c0der** [comment]
@AnthonyMDev can you please review this PR? It seems like the change you've requested is done.

**Comment by AnthonyMDev** [comment]
Sorry, yeah this looks good!

**Pull request merged**